lib/chef/guard_interpreter/resource_guard_interpreter.rb

```diff
@@ -53,33 +62,24 @@ def evaluate
         # the one attribute that causes this changes its default to be
         # the same after some period to prepare for deprecation
         if @resource.class.respond_to?(:get_default_attributes)
-          block_attributes = @resource.class.send(:get_default_attributes, @command_opts).merge(block_attributes)
+          @resource.class.send(:get_default_attributes).each do |attribute, value|
+            @resource.send(attribute, value)
+          end
         end
 
-        resource_block = block_from_attributes(block_attributes)
-        evaluate_action(nil, &resource_block)
-      end
```

Comment:
`evaluate` is the entry point for all the behavior in this class, and it:
* builds a hash of parameters to be set on the guard resource
* makes a proc that sets the parameters on the guard resource
* passes the proc to another function that `instance_eval`s the proc on the guard resource

Instead it can set the parameters on the guard resource directly and avoid all the indirection.